src/solver/optim/api/include/antares/solver/optim/api/linearProblemData.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace Antares::Solver::Optim::Api
+{
+
+class LinearProblemData
+{
+public:
+    unsigned getTimeResolutionInMinutes();
```

Comment:
Question on **getTimeResolutionInMinutes** : 
Why **in minutes** ?
The smallest time granularity in Antares Simulator is hour, so I can't see a usage for a time amount expressed in minutes

Reply:
I think we better not limit it to the hour right now

Reply:
Well, as principles **YAGNI** and **KISS**, we should not make assumptions for what will come in the future.
Our concern should be to make the current code clear.
Changes will come in time, if and when we need them.

Reply:
limiting to the hour would also be making an assumption, because the API is supposed to allow future use cases that are not possible today in antares. and it would be more painful to change back, to allow minutes. the other way around (the current version) allows setting the granularity to 1 minute or 1 hour, has no overcost, and is more future-proof.

Reply:
The shortest time interval in Antares is the hour, that's not an assumption, but a fact.
Now, what will happen if we leave this **getTimeResolutionInMinute** as it is and there is no such minutes in Antares Simulator ? Well it's just confusing, and it will certainly make someone waste time.

Besides, changing back the name of a function at some point and because we need it is not time consuming.